### PR TITLE
docs: Add Using Kakao Login JS SDK

### PR DESCRIPTION
**Pull request by MiryangJung**
## What kind of change does this PR introduce?

Add docs Using Kakao Login JS SDK

## Additional context

https://github.com/supabase/gotrue/pull/1381#issuecomment-1940693055

### CODE REVIEW

**Comment by MiryangJung** [comment]
If someone is experiencing difficulties, please refer to [this](https://miryang.dev/blog/how-to-use-kakao-login-js-sdk-with-supabase) article. and Please let me know if someone would like me to write the blog post in English.

File: apps/docs/content/guides/auth/social-login/auth-kakao.mdx [hunk 1]
```diff
@@ -132,6 +132,51 @@ suspend fun signOut() {
...
+```
+const res = await auth.signInWithIdToken({
+  provider: 'kakao',
+  token: id_token,
+});
+```
+
+### Configuration
+
+1. You must set 'State' to 'ON' under [OpenID Connect Activation](https://developers.kakao.com/docs/latest/en/kakaologin/prerequisite#activate-oidc) on Kakao Developers portal Application Dashboard.
```
Comment (J0) [inline apps/docs/content/guides/auth/social-login/auth-kakao.mdx:177]: /nit consider using imperative
```suggestion
1. Set 'State' to 'ON' under [OpenID Connect Activation](https://developers.kakao.com/docs/latest/en/kakaologin/prerequisite#activate-oidc) on Kakao Developers portal Application Dashboard.
```

File: apps/docs/content/guides/auth/social-login/auth-kakao.mdx [hunk 2]
```diff
@@ -132,6 +132,51 @@ suspend fun signOut() {
...
+2. You must add `openid` to [scope](https://developers.kakao.com/docs/latest/en/kakaologin/common#additional-consent-scope) along with the scope values you want to obtain consent.
```
Comment (J0) [inline apps/docs/content/guides/auth/social-login/auth-kakao.mdx:178]:
```suggestion
2. Add `openid` to [scope](https://developers.kakao.com/docs/latest/en/kakaologin/common#additional-consent-scope) along with the scope values you wish to obtain consent for.
```

File: apps/docs/content/guides/auth/social-login/auth-kakao.mdx [hunk 3]
```diff
@@ -132,6 +132,51 @@ suspend fun signOut() {
...
+      code,
+      client_secret: '<CLIENT_SECRET>',
+    }),
+  });
+
+  const {id_token} = await res.json();
+}
+```
+
+You can signin with the ID token you got.
```
Comment (J0) [inline apps/docs/content/guides/auth/social-login/auth-kakao.mdx:166]:
```suggestion
Use the ID Token to sign in:
```

File: apps/docs/content/guides/auth/social-login/auth-kakao.mdx [hunk 4]
```diff
@@ -132,6 +132,51 @@ suspend fun signOut() {
 </TabPanel>
 </Tabs>
 
+## Using Kakao Login JS SDK
+
+[Kakao Login JS SDK](https://developers.kakao.com/docs/latest/en/kakaologin/js) is an official Kakao SDK for authenticating Kakao users on websites.
+
+You need a code to get [ID Token with the issued authorization code](https://developers.kakao.com/docs/latest/en/kakaologin/common#login-with-oidc).
```
Comment (J0) [inline apps/docs/content/guides/auth/social-login/auth-kakao.mdx:139]:
```suggestion
Exchange the [authorization code returned by Kakao API](https://developers.kakao.com/docs/latest/en/kakaologin/rest-api#request-code) for an [ID Token](https://developers.kakao.com/docs/latest/en/kakaologin/common#login-with-oidc). 
```

**Review by J0 — APPROVED** [review]
Hey I left some suggestions / nits for your consideration. Hope you don't mind.

Content wise it looks great, thank you so much for taking the time to put this together and sorry it took so long for us to get back to you.

@charislam feel free to add on if you have any suggestions

**Comment by charislam** [comment]
Awesome thanks @MiryangJung! Can you just run a quick `npm run format` so it will pass our pre-merge checks?

**Comment by MiryangJung** [comment]
> Awesome thanks @MiryangJung! Can you just run a quick `npm run format` so it will pass our pre-merge checks?

yes!

**Pull request merged**